pkg/doc/verifiable/credential_jwt_proof_test.go

```diff
@@ -0,0 +1,196 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package verifiable
+
+import (
+	"errors"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/square/go-jose/v3/jwt"
+	"github.com/stretchr/testify/require"
+)
+
+const certPrefix = "testdata/crypto"
+
+const jwtTestCredential = `
+{
+	"@context": [
+	  "https://www.w3.org/2018/credentials/v1",
+	  "https://www.w3.org/2018/credentials/examples/v1"
+	],
+	"type": ["VerifiableCredential", "UniversityDegreeCredential"],
+	"credentialSubject": {
+	  "id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
+	  "degree": {
+		"type": "BachelorDegree",
+		"university": "MIT"
+	  }
+	},
+  "issuer": {
+    "id": "did:example:76e12ec712ebc6f1c221ebfeb1f",
+    "name": "Example University"
+  },
+  "issuanceDate": "2010-01-01T19:23:24Z",
+  "expirationDate": "2020-01-01T19:23:24Z"
+}
+`
+
+const keyID = "1"
+
+func TestNewCredentialFromJWTSigned(t *testing.T) {
+	testCred := []byte(jwtTestCredential)
+
+	keyFetcher := createKeyFetcher(t)
+
+	t.Run("jwt encoding with fields minimization", func(t *testing.T) {
+		vcFromJWT, err := NewCredential(
+			createJWT(t, testCred, false),
+			WithJWTPublicKeyFetcher(keyFetcher))
+
+		require.NoError(t, err)
+
+		vcFromPlain, err := NewCredential(testCred)
+		require.NoError(t, err)
+
+		require.Equal(t, vcFromPlain, vcFromJWT)
+	})
+
+	t.Run("jwt encoding with minimized fields", func(t *testing.T) {
+		vcFromJWT, err := NewCredential(
+			createJWT(t, testCred, true),
+			WithJWTPublicKeyFetcher(keyFetcher))
+
+		require.NoError(t, err)
+
+		vcFromPlain, err := NewCredential(testCred)
+		require.NoError(t, err)
+
+		require.Equal(t, vcFromPlain, vcFromJWT)
+	})
+
+	t.Run("Failed JWT signature verification", func(t *testing.T) {
+		_, err := NewCredential(
+			createJWT(t, testCred, true),
+
+			// passing holder's key, while expecting issuer one
+			WithJWTPublicKeyFetcher(func(issuerID, keyID string) (interface{}, error) {
+				publicKey, err := readPublicKey(filepath.Join(certPrefix, "holder_public.pem"))
+				require.NoError(t, err)
+				require.NotNil(t, publicKey)
+
+				return publicKey, nil
+			}))
+
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "JWT decoding failed")
```

Comment:
use `require.EqualError(t, e, ...)` instead

Reply:
This is intentional - I'm checking only a part of the error message here...